testsuite/MDAnalysisTests/analysis/test_diffusionmap.py

```diff
@@ -20,57 +20,70 @@
 #
 from __future__ import print_function, absolute_import
-from unittest import TestCase
-import numpy as np
+
 import MDAnalysis
 import MDAnalysis.analysis.diffusionmap as diffusionmap
-from numpy.testing import (assert_almost_equal, assert_equal,
-                           assert_array_almost_equal,raises)
+import numpy as np
+import pytest
+from MDAnalysisTests.datafiles import PDB, XTC
+from numpy.testing import assert_array_almost_equal
 
 
-from MDAnalysisTests.datafiles import PDB, XTC
+class TestDiffusionmap(object):
+    @staticmethod
+    @pytest.fixture()
+    def u():
+        return MDAnalysis.Universe(PDB, XTC)
+
+    @staticmethod
+    @pytest.fixture()
+    def dist(u):
+        return diffusionmap.DistanceMatrix(u, select='backbone')
+
+    @staticmethod
+    @pytest.fixture()
+    def dmap(dist):
+        d_map = diffusionmap.DiffusionMap(dist)
+        d_map.run()
+        return d_map
 
+    @staticmethod
+    @pytest.fixture()
+    def eigvals(dmap):
+        return dmap.eigenvalues
 
-class TestDiffusionmap(TestCase):
-    def setUp(self):
-        self.u = MDAnalysis.Universe(PDB, XTC)
-        self.dist = diffusionmap.DistanceMatrix(self.u, select='backbone')
-        self.dmap = diffusionmap.DiffusionMap(self.dist)
-        self.dmap.run()
-        self.eigvals = self.dmap.eigenvalues
-        self.eigvects = self.dmap._eigenvectors
+    @staticmethod
+    @pytest.fixture()
+    def eigvects(dmap):
+        return dmap._eigenvectors
 
-    def test_eg(self):
+    def test_eg(self, eigvals, dist):
         # number of frames is trajectory is now 10 vs. 98
-        assert_equal(self.eigvals.shape, (self.dist.n_frames, ))
+        assert eigvals.shape == (dist.n_frames,)
         # makes no sense to test values here, no physical meaning
 
-    def test_dist_weights(self):
-        backbone = self.u.select_atoms('backbone')
+    def test_dist_weights(self, u):
+        backbone = u.select_atoms('backbone')
         weights_atoms = np.ones(len(backbone.atoms))
-        self.dist = diffusionmap.DistanceMatrix(self.u, select='backbone',
-                                                weights=weights_atoms,
-                                                step=3)
-        self.dist.run()
-        self.dmap = diffusionmap.DiffusionMap(self.dist)
-        self.dmap.run()
-        assert_array_almost_equal(self.dmap.eigenvalues, [1,1,1,1], 4)
-        assert_array_almost_equal(self.dmap._eigenvectors,
-                                  ([[ 0, 0, 1, 0],
-                                    [ 0, 0, 0, 1],
-                                    [ -.707,-.707, 0, 0],
-                                    [  .707,-.707, 0, 0]]) ,2)
+        dist = diffusionmap.DistanceMatrix(u, select='backbone', weights=weights_atoms, step=3)
+        dist.run()
+        dmap = diffusionmap.DiffusionMap(dist)
+        dmap.run()
+        assert_array_almost_equal(dmap.eigenvalues, [1, 1, 1, 1], 4)
```

Comment:
I don' think so. The value of `dist` is different here. It has `weights=weights_atoms, step=3` which isn't in the `dist` fixture that is used by the `dmap` fixture.